### ISSUE TITLE
trouble reproducing h2 wavefunction optimization example with PySCF?

### ISSUE THREAD

**Issue by currymj** [NLESC-JCER/QMCTorch#50]
I'm currently trying to reproduce the CPU wavefunction optimization example.

I don't have a license for `scm`, so can't use the `adf` calculator. The only change I made to `h2.py` was to replace `adf` with `pyscf`. But when I run the example, I don't get a nice plot of variance dropping over time, as shown in the documentation. Instead, I get a chart that looks like the following:

![image](https://user-images.githubusercontent.com/3802732/86279491-19349e00-bba8-11ea-997f-d197fca78973.png)

It is totally possible that this is user error, or that there are some other changes that need to be made for this to work with pyscf.

Also, if the software is currently in a work-in-progress state such that reports like this from end users are unhelpful, please let me know.

**Comment by NicoRenaud** [comment]
Hey Michael, 

thanks for your input and sorry for the delay I was taking a break from the project. The code is very much in development still and I should make that more clear somewhere. 

The issue with pyscf is that the GTO won't have the correct cusp condition and therefore the gradients are all wrong. This lead to the optimization curve you obtained. 

There are two solutions to fix that either introduce pseudopotential to remove part of the the 1/r dependency in the electron nuclei potential or refit the GTO to STO. I'm planning to implement the second one as soon as I have time. 

Thanks again for your input !

**Comment by NicoRenaud** [comment]
I've merged into the master a new method called `gto2sto` in the `wf_orbital` class that  fits the contracted GTOs used by pyscf into STOs. It can be used as follows: 

```python
wf = Orbital(mol).gto2sto()
``` 

The resulting instance can then be used for optimization. 
I've tried it for the wave function optimization of CH4 molecule and it seemed to work fine. At least better than the original GTOs even though the variance of the local energies seem large.

Let me know if that works for you of you try it out !
